src/types/local_date.rs

```diff
@@ -0,0 +1,290 @@
+        })
+    }
+
+    #[napi(js_name = "toString")]
+    pub fn to_format(&self) -> String {
+        self.to_string()
+    }
+
+    pub fn get_cql_date(&self) -> CqlDate {
+        CqlDate(((1 << 31) + self.value) as u32)
+    }
+
+    pub fn from_cql_date(date: CqlDate) -> Self {
+        let value: i32 = date.0 as i32 - (1 << 31);
+        let date = Ymd::from_days(value.into())
+            .map(|x| (Some(x.year), Some(x.month), Some(x.day)))
+            .unwrap_or((None, None, None));
+        LocalDateWrapper {
+            value,
+            year: date.0,
+            month: date.1,
+            day: date.2,
+            in_date: (MIN_JS_DATE..=MAX_JS_DATE).contains(&value),
+        }
+    }
+
+    /// Returns the number of days since 01.01.1970 based on a String representing the date.
+    #[napi]
+    pub fn from_string(value: String) -> napi::Result<i32> {
+        match value.chars().filter(|c| *c == '-').count() {
+            d if d < 2 => match value.parse::<i32>() {
+                Ok(val) => Ok(val),
+                Err(_) => Err(js_error("Invalid format of string")),
+            },
+            2 | 3 => {
+                let lambda = |s: String| -> Result<(i32, i8, i8), ParseIntError> {
+                    let mut cut: usize = 0;
+                    if s.starts_with('-') {
+                        cut = 1;
+                    }
+
+                    let mut parts = s.get(cut..).unwrap().split('-');
+                    let year = parts.next().unwrap().parse::<i32>()?;
+                    Ok((
+                        match cut {
+                            0 => year,
+                            _ => -year,
+                        },
+                        parts.next().unwrap().parse::<i8>()?,
+                        parts.next().unwrap().parse::<i8>()?,
+                    ))
+                };
+
+                match lambda(value) {
+                    Ok(s) => {
+                        let date = Ymd {
+                            year: s.0,
+                            month: s.1,
+                            day: s.2,
+                        };
+                        Ok(date.to_days())
+                    }
+                    Err(_) => Err(js_error("Invalid parse of string")),
+                }
+            }
+            _ => Err(js_error("Invalid format of string")),
+        }
+    }
+}
+
+impl fmt::Display for LocalDateWrapper {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.in_date {
+            if self.year.unwrap() < 0 {
+                let year_string = (self.year.unwrap().unsigned_abs()).to_string();
+                write!(
+                    f,
+                    "-{}{}",
+                    "0".repeat(max(4 - year_string.len() as i8, 0) as usize),
+                    year_string
+                )?;
+            } else {
+                let year_string = self.year.unwrap().to_string();
```

Comment:
:wrench: This is an allocation in `Display` - it's unnacceptable again.

Reply:
Instead, you should pass `year` to `write!` directly, because then its `Display` impl will be employed without allocations.